libs/shared/devops-copilot/feature/src/lib/devops-copilot-panel/use-thread.tsx

```diff
@@ -0,0 +1,84 @@
+import { useAuth0 } from '@auth0/auth0-react'
+import { useEffect, useState } from 'react'
+import { type Thread } from './devops-copilot-panel'
+import { HACKATHON_API_BASE_URL } from './submit-message'
+
+interface UseCurrentThreadOptions {
+  organizationId: string
+  threadId?: string
+}
+
+interface UseCurrentThreadResult {
+  thread: Thread
+  setThread: (thread: Thread) => void
+  isLoading: boolean
+  error: Error | null
+}
+
+export async function fetchThread(userSub: string, organizationId: string, threadId: string, token: string) {
+  const response = await fetch(
+    `${HACKATHON_API_BASE_URL}/owner/${userSub}/organization/${organizationId}/thread/${threadId}`,
+    {
+      headers: {
+        Authorization: `Bearer ${token}`,
+      },
+    }
+  )
+
+  if (!response.ok) {
+    throw new Error('Failed to fetch thread')
+  }
+
+  const data = await response.json()
+  return data.results || []
+}
```

Comment:
This could be extracted to its own file. In the same fashion as it has been done for `submit-vote` and `submit-message`